### ISSUE TITLE
Problem encoding large values in long

### ISSUE THREAD

**Issue by laslowh** [flavray/avro-rs#77]
This code appears to fail the test. It's a round trip encoding/decoding of a large integer value. The value is larger than an 2^32 but smaller than 2^63.

```
    pub const JSON: &str = r#"
    { "name": "athing", "type": "long" }
    "#;

    fn serialize_round_trip2() {
        let json_scheme: serde_json::Value = serde_json::from_str(JSON).unwrap();
        let scheme = Schema::parse(&json_scheme).unwrap();
        let a_thing = 1553201383562400 as i64;
        let bytes = to_avro_datum(&scheme, to_value(a_thing).unwrap()).unwrap();
        let rehy: i64 =
            from_value(&from_avro_datum(&scheme, &mut &bytes[..], None).unwrap())
                .unwrap();
        assert_eq!(rehy, 1553201383562400 as i64);
    }
```

When run with `cargo test` the result is:

```
---- tests::serialize_round_trip2 stdout ----
thread 'tests::serialize_round_trip2' panicked at 'assertion failed: `(left == right)`
  left: `-1553201383790593`,
 right: `1553201383562400`', src/lib.rs:502:9
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

**Comment by laslowh** [comment]
I think maybe this is the problem, although I have to admit I don't really understand what that code is doing.

```
diff --git a/src/util.rs b/src/util.rs
index f3b5ee4..3d14571 100644
--- a/src/util.rs
+++ b/src/util.rs
@@ -68,7 +68,7 @@ pub fn zig_i32(n: i32, buffer: &mut Vec<u8>) {
 }
 
 pub fn zig_i64(n: i64, buffer: &mut Vec<u8>) {
-    encode_variable((n << 1) ^ (n >> 31), buffer)
+    encode_variable((n << 1) ^ (n >> 63), buffer)
 }
 
 pub fn zag_i32<R: Read>(reader: &mut R) -> Result<i32, Error> {

```

**Comment by flavray** [comment]
I tried to reproduce with the latest version (`0.6.5`) and was not able to. The code compiles correctly and this new test passes without panicking.
What version of the library are you running?

**Comment by laslowh** [comment]
Version 0.6.0.  I will try against the latest. Thanks.

**Comment by laslowh** [comment]
Yes, you're correct, it was fixed by 9492d6c38a3ef6206ec0a82192a08a493864fe6a sorry for the trouble, closing.
